src/components/designer/taro/actions/ChangeTaroBackendModal.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from 'react';
+import { useAsyncCallback } from 'react-async-hook';
+import { SwapOutlined } from '@ant-design/icons';
+import styled from '@emotion/styled';
+import { Alert, Col, Form, Modal, Row } from 'antd';
+import { usePrefixedTranslation } from 'hooks';
+import { Status } from 'shared/types';
+import { useStoreActions, useStoreState } from 'store';
+import { Network } from 'types';
+import LightningNodeSelect from 'components/common/form/LightningNodeSelect';
+import TaroNodeSelect from 'components/common/form/TaroNodeSelect';
+
+const Styled = {
+  IconCol: styled(Col)`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  `,
+  Restart: styled.p`
+    font-style: italic;
+    opacity: 0.6;
+  `,
+};
+
+interface Props {
+  network: Network;
+}
+
+const ChangeTaroBackendModal: React.FC<Props> = ({ network }) => {
+  const { l } = usePrefixedTranslation(
+    'cmps.designer.lightning.actions.ChangeTaroBackendModal',
+  );
+  const [form] = Form.useForm();
+  const [compatWarning, setCompatWarning] = useState<string>();
+  const {
+    visible,
+    lndName: LNDName,
+    taroName,
+  } = useStoreState(s => s.modals.changeTaroBackend);
+  const [selectedTaro, setSelectedTaro] = useState(taroName);
+  const [selectedLNDBackend, setSelectedLNDBackend] = useState(LNDName);
+  const { dockerRepoState } = useStoreState(s => s.app);
+  const { hideChangeTaroBackend } = useStoreActions(s => s.modals);
+  const { updateTaroBackendNode } = useStoreActions(s => s.network);
+  const { notify } = useStoreActions(s => s.app);
+
+  const changeAsync = useAsyncCallback(async (taroName: string, LNDName: string) => {
+    try {
+      await updateTaroBackendNode({
+        id: network.id,
+        taroName,
+        lndName: LNDName,
+      });
+      notify({
+        message: l('successTitle'),
+        description: l('successDesc', { taroName, LNDName }),
+      });
+      hideChangeTaroBackend();
+    } catch (error: any) {
+      notify({ message: l('submitError'), error });
+    }
+  });
+
+  useEffect(() => {
+    const { lightning, taro } = network.nodes;
+    const taroNode = taro.find(n => n.name === selectedTaro);
+    const lndBackend = lightning.find(n => n.name === selectedLNDBackend);
+    if (taroNode && lndBackend) {
+      const { compatibility } = dockerRepoState.images[taroNode.implementation];
+      if (compatibility) {
+        const requiredVersion = compatibility[lndBackend.version];
+        //version compatibility function breaks down
+        if (!requiredVersion) {
```

Comment:
I had a lot of trouble hitting this if statement, I couldn't figure out how what test you were running where your coverage would clear.